pkg/app/piped/platformprovider/kubernetes/loader.go

```diff
@@ -173,8 +173,8 @@ func setNamespace(manifests []Manifest, namespace string) {
 	if namespace == "" {
 		return
 	}
-	for _, m := range manifests {
-		m.Key.Namespace = namespace
+	for i := range manifests {
+		manifests[i].Key.Namespace = namespace
```

Comment:
I could understand what this wants to do, but originally the manifest is not a pointer, right?

Reply:
basically, this function accepts `manifests` which of type slice as it input. Go makes all function's input pass by value, thus it copy this manifests slide and manipulate thing inside the function. The point to note here is that even it copy the slide, the underlying array of that slide  is remained to be the same, so basically change "that" array will change after function returns. Previously, instead of change the underlying array, `m` is a temporary var that ref to element of slide and change make on it does not reflect on the underlying array thus the changes are not remained after function end.

https://go.dev/play/p/Osr6eY4eh0A

Reply:
Oh, I totally understand. Thank you.